quickwit/quickwit-search/src/collector.rs

```diff
@@ -293,25 +233,64 @@ enum AggregationSegmentCollectors {
 pub struct QuickwitSegmentCollector {
     num_hits: u64,
     split_id: String,
-    score_extractor: SortingFieldExtractorPair,
+    score_extractors: Vec<SortingFieldExtractorComponent>,
     // PartialHits in this heap don't contain a split_id yet.
     top_k_hits: TopK<PartialHit, PartialHitSortingKey, HitSortingMapper>,
     segment_ord: u32,
     timestamp_filter_opt: Option<TimestampFilter>,
     aggregation: Option<AggregationSegmentCollectors>,
+    search_after: Option<PartialHit>,
 }
 
 impl QuickwitSegmentCollector {
     #[inline]
     fn collect_top_k(&mut self, doc_id: DocId, score: Score) {
-        let (sort_value, sort_value2) =
-            self.score_extractor.extract_typed_sort_value(doc_id, score);
+        let sort_values: Vec<_> = self
+            .score_extractors
+            .iter()
+            .map(|extractor| extractor.extract_typed_sort_value_opt(doc_id, score))
```

Comment:
No allocation in this function please. this is called once per document match by the query.